src/include/function/hash/hash_functions.h

```diff
@@ -54,99 +56,178 @@ struct CombineHash {
     result = murmurhash64(key);
 }
 
 template<>
-inline void Hash::operation(const int8_t& key, common::hash_t& result) {
+inline void Hash::operation(
+    const int8_t& key, common::hash_t& result, common::ValueVector* /*keyVector*/) {
     result = murmurhash64(key);
 }
 
 template<>
-inline void Hash::operation(const common::int128_t& key, common::hash_t& result) {
+inline void Hash::operation(
+    const common::int128_t& key, common::hash_t& result, common::ValueVector* /*keyVector*/) {
     result = murmurhash64(key.low) ^ murmurhash64(key.high);
 }
 
 template<>
-inline void Hash::operation(const double& key, common::hash_t& result) {
+inline void Hash::operation(
+    const double& key, common::hash_t& result, common::ValueVector* /*keyVector*/) {
     result = std::hash<double>()(key);
 }
 
 template<>
-inline void Hash::operation(const float& key, common::hash_t& result) {
+inline void Hash::operation(
+    const float& key, common::hash_t& result, common::ValueVector* /*keyVector*/) {
     result = std::hash<float>()(key);
 }
 
 template<>
-inline void Hash::operation(const std::string& key, common::hash_t& result) {
+inline void Hash::operation(
+    const std::string& key, common::hash_t& result, common::ValueVector* /*keyVector*/) {
     result = std::hash<std::string>()(key);
 }
 
 template<>
-inline void Hash::operation(const std::string_view& key, common::hash_t& result) {
+inline void Hash::operation(
+    const std::string_view& key, common::hash_t& result, common::ValueVector* /*keyVector*/) {
     result = std::hash<std::string_view>()(key);
 }
 
 template<>
-inline void Hash::operation(const common::ku_string_t& key, common::hash_t& result) {
+inline void Hash::operation(
+    const common::ku_string_t& key, common::hash_t& result, common::ValueVector* /*keyVector*/) {
     result = std::hash<std::string_view>()(key.getAsStringView());
 }
 
 template<>
-inline void Hash::operation(const common::interval_t& key, common::hash_t& result) {
+inline void Hash::operation(
+    const common::interval_t& key, common::hash_t& result, common::ValueVector* /*keyVector*/) {
     result = combineHashScalar(murmurhash64(key.months),
         combineHashScalar(murmurhash64(key.days), murmurhash64(key.micros)));
 }
 
 template<>
-inline void Hash::operation(const common::ku_list_t& /*key*/, common::hash_t& /*result*/) {
-    throw common::RuntimeException(
-        "Computing hash value of list DataType is currently unsupported.");
+inline void Hash::operation(
+    const common::list_entry_t& key, common::hash_t& result, common::ValueVector* keyVector) {
+    auto dataVector = common::ListVector::getDataVector(keyVector);
+    result = NULL_HASH;
+    common::hash_t tmpResult;
+    for (auto i = 0u; i < key.size; i++) {
+        auto pos = key.offset + i;
+        if (dataVector->isNull(pos)) {
+            result = combineHashScalar(result, NULL_HASH);
+        } else {
+            switch (dataVector->dataType.getPhysicalType()) {
```

Comment:
Check if you can use `TypeUtils::visit` to simplify the switch/case code here.

Reply:
Plus let's have decent coverage on this code path.